### DISCUSSION TITLE
Any examples of mpr in version 5.0?

### DISCUSSION THREAD

**Discussion by jpadkw** [fo-dicom/fo-dicom#1275] [Q&A]
I know that mpr was applied in version 5.0.

Any examples of mpr in version 5.0?

Or, please advise on how to apply it.

Thank you.

**Comment by gofal** [comment]
There are some helper-classes to do some reconstruction in backend. Since fo-dicom is a backend class library there is no direct support for fast GPU-based realtime MPR. But you can do the following:

if you have some files or DicomDataset of slices, then create a new instance of FellowOakDicom.imaging.Reconstruction.ImageData for each of that files or datasets. you can then check that data and filter out (e.g. there is a scount image of different orientation in the stack of images). Then you pass the IEnumerable<ImageData> to the constrctor of FellowOakDicom.Imaging.Reconstruction.VolumeData.
This VolumeData then gives you some properties like the Spacings or the bounding box. 

You can then calculate single slices out of this volume by calling new Slice (volume, PointTopLeft, vectorrow, vectorcol, rows, cols, spacing). Or you can generate a whole stack of sclices by calling new Stack(volume, orientation, spacing, slicedistance). 

the data of the slice can then be copied into an byte array by calling slice.RenderIntoByteArray. Of if you have added the nuget-packages fo-dicom.Imaging.Desktop or fo-dicom.Imaging.ImageSharp, then you have extension methods available: slide.AsBitmap(), slide.AsWpfBitmap() r slice.AsImage(),
